src/unittests/f32_interval.spec.ts

```diff
@@ -3314,3 +3316,58 @@ interface RefractCase {
       );
     });
 }
+
+interface ModfCase {
+  input: number;
+  fract: IntervalBounds;
+  whole: IntervalBounds;
+}
+
+g.test('modfInterval')
+  .paramsSubcasesOnly<ModfCase>(
+    // When performing modf(x), where x is an integer value, the acceptance intervals are somewhat unintuitive.
+    // Specifically the acceptance intervals will be something like { fract: [0, 1], whole: [x-1, x] } instead of just
+    // { fract: [0], whole: [x] }.
+    // This is because the accuracy calculations for `%`, which is used for calculating the results of modf, has
+    // trunc(y/x) in it, which introduces 2.5 ULP of error before the truncate. So when near the integer boundary, this
```

Comment:
The `trunc(y/x)` comes from `modf`'s description of `(e % 1.0)`, and `e1 % e2` is described as `e1 - e2 * trunc(e1 / e2)` correct?

I'm not sure if this is supposed to reflect the precision of the builtin, or whether it was just the simplest way of describing the behaviour for negatives.

In https://www.w3.org/TR/WGSL/#floating-point-accuracy, `modf()` is only listed as being `Correctly rounded`.

@dneto0 , @alan-baker - should we really be accounting for the ULP of the divide of `trunc(x / 1.0)` here?

Reply:
Talked with @dneto0 about this offline, you are correct, this shouldn't be done using inherited accuracy. I have updated the implementation and tests.